galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -140,22 +157,33 @@ func (d *AnalyzingDistributor) analyzeAndDistribute(cancelCh chan struct{}, name
 	d.s.Analyzer.Analyze(ctx)
 	scope.Analysis.Debugf("Finished analyzing the current snapshot, found messages: %v", ctx.messages)
 
-	// Only keep messages for resources in namespaces we want to analyze if the
-	// message doesn't have an origin (meaning we can't determine the
-	// namespace). Also kept are cluster-level resources where the namespace is
-	// the empty string. If no such limit is specified, keep them all.
 	var msgs diag.Messages
-	if len(namespaces) == 0 {
-		msgs = ctx.messages
-	} else {
-		for _, m := range ctx.messages {
-			if m.Origin != nil && m.Origin.Namespace() != "" {
-				if _, ok := namespaces[m.Origin.Namespace()]; !ok {
-					continue
-				}
+FilterMessages:
+	for _, m := range ctx.messages {
```

Comment:
It wouldn't hurt to extract this whole filter loop into a function, just to keep `analyzeAndDistribute` easier to read.

Reply:
Done.